contracts/viewers/FLIRebalanceViewer.sol

```diff
@@ -0,0 +1,258 @@
+        return outputAmount.preciseDiv(_sellSize);
+    }
+
+    /**
+     * Gets the ordered priority of which exchanges to use for a rebalance
+     *
+     * @param _uniswapV3Price               price of rebalance trade on Uniswap V3
+     * @param _uniswapV2Price               price of rebalance trade on Uniswap V2
+     * @param _actionInfo                   ActionInfo struct
+     *
+     * @return string[] memory              Ordered array of exchange names to use. Earlier elements in the array produce the best trades
+     * @return ShouldRebalance[] memory     Array of ShouldRebalance Enums. Ordered relative to returned exchange names array
+     */
+    function _getExchangePriority(
+        uint256 _uniswapV3Price,
+        uint256 _uniswapV2Price,
+        ActionInfo memory _actionInfo
+    )
+        internal
+        view
+        returns (string[2] memory, FlexibleLeverageStrategyExtension.ShouldRebalance[2] memory)
+    {
+
+        // If no rebalance is required, set price to 0 so it is ordered last
+        if (_actionInfo.rebalanceActions[_actionInfo.uniV3Index] == FlexibleLeverageStrategyExtension.ShouldRebalance.NONE) _uniswapV3Price = 0;
+        if (_actionInfo.rebalanceActions[_actionInfo.uniV2Index] == FlexibleLeverageStrategyExtension.ShouldRebalance.NONE) _uniswapV2Price = 0;
+
+        if (_uniswapV3Price > _uniswapV2Price) {
+            return ([ uniswapV3ExchangeName, uniswapV2ExchangeName ],
+                    [ _actionInfo.rebalanceActions[_actionInfo.uniV3Index], _actionInfo.rebalanceActions[_actionInfo.uniV2Index] ]);
+        } else {
+            return ([ uniswapV2ExchangeName, uniswapV3ExchangeName ],
+                    [ _actionInfo.rebalanceActions[_actionInfo.uniV2Index], _actionInfo.rebalanceActions[_actionInfo.uniV3Index] ]);
+        }
+    }
+
+    /**
+     * Creates the an ActionInfo struct containing information about the rebalancing action. Note: chunkSendQuantity is
+     * ordered [V3, V2], so there is no need to use uniV3Index or uniV2Index when accessing it.
+     *
+     * @param _minLeverage          Min leverage ratio
+     * @param _maxLeverage          Max leverage ratio
+     *
+     * @return actionInfo           Populated ActionInfo struct
+     */
+    function _getActionInfo(uint256 _minLeverage, uint256 _maxLeverage) internal view returns (ActionInfo memory actionInfo) {
+
+        (actionInfo.exchangeNames, actionInfo.rebalanceActions) = fliStrategyExtension.shouldRebalanceWithBounds(
+            _minLeverage,
+            _maxLeverage
+        );
+
+        (actionInfo.uniV3Index, ) = actionInfo.exchangeNames.indexOf(uniswapV3ExchangeName);
+        (actionInfo.uniV2Index, ) = actionInfo.exchangeNames.indexOf(uniswapV2ExchangeName);
+
+        actionInfo.minLeverage = _minLeverage;
+        actionInfo.maxLeverage = _maxLeverage;
+
+        string[] memory exchangeNames = new string[](2);
```

Comment:
Just noticed this, we can delete this variable and use `actionInfo.exchangeNames` directly right

Reply:
We could use that, but then we'd have to refactor to use uniV3Index and uniV2Index when accessing chunkSendQuantiy. Both ways will work though. Should I make the change?